cogs/tempchannels/tempchannels.py

```diff
@@ -536,37 +544,42 @@ async def tempChannelsDenyRemove(self, ctx: Context, *, role: discord.Role):
-                # Channel created, see when we should delete it.
-                try:
-                    chanObj = self.bot.get_channel(guildData[KEY_CH_ID])
-                    await chanObj.delete()
-                except discord.DiscordException:
-                    self.logger.error("Could not delete channel!", exc_info=True)
-                    await ctx.send(
-                        ":warning: TempChannel: Something went wrong "
-                        "while trying to delete the channel. Please "
-                        "check the console log for details."
-                    )
-                else:
-                    guildData[KEY_CH_ID] = None
-                    guildData[KEY_CH_CREATED] = False
-                    self.logger.info(
-                        "%s (%s) deleted the temp channel #%s (%s) in %s (%s).",
-                        ctx.author.name,
-                        ctx.author.id,
-                        chanObj.name,
-                        chanObj.id,
-                        ctx.guild.name,
-                        ctx.guild.id,
-                    )
-                    await ctx.send(":white_check_mark: TempChannel: Channel deleted")
-            else:
+        guildConfig = self.config.guild(ctx.guild)
+        channelCreated = await guildConfig.get_attr(KEY_CH_CREATED)()
+        channelId = await guildConfig.get_attr(KEY_CH_ID)()
+
+        if channelCreated and channelId:
+            # Channel created, see when we should delete it.
+            try:
+                chanObj = self.bot.get_channel(channelId)
+                await chanObj.delete()
+            except discord.DiscordException:
+                self.logger.error("Could not delete channel!", exc_info=True)
                 await ctx.send(
-                    ":negative_squared_cross_mark: TempChannel: There is no "
-                    "temporary channel to delete!"
+                    ":warning: TempChannel: Something went wrong "
+                    "while trying to delete the channel. Please "
+                    "check the console log for details."
                 )
+            else:
+                channelId = None
+                channelCreated = False
+                self.logger.info(
+                    "%s (%s) deleted the temp channel #%s (%s) in %s (%s).",
+                    ctx.author.name,
+                    ctx.author.id,
+                    chanObj.name,
+                    chanObj.id,
+                    ctx.guild.name,
+                    ctx.guild.id,
+                )
+                await ctx.send(":white_check_mark: TempChannel: Channel deleted")
+        else:
+            await ctx.send(
+                ":negative_squared_cross_mark: TempChannel: There is no "
+                "temporary channel to delete!"
+            )
+        await guildConfig.get_attr(KEY_CH_CREATED).set(channelCreated)
+        await guildConfig.get_attr(KEY_CH_ID).set(channelId)
```

Comment:
Can we save this directly in the `else` clause above?

The reason is because here and in other places (but I think particularly in this case), the behaviour changes slightly. Previously, when using `all()`, when we adjust the attribute (in this case `channelId` and `channelCreated`), it will get saved regardless of what else happens in this method.

Suppose in the try/except/else, specifically in the else case, we hit an Exception in the `ctx.send`. At this point, because we've hit the `else` block, we we able to delete the channel. Since we hit an Exception, we bail early, but don't actually save the delete back into Config.

Reply:
Oh yes. I could see the `try/except` block. Should have been more careful when working on this method.